### PR TITLE
Coredump reporter

### PR DESCRIPTION
**Pull request by freva**
* Copies `CoredumpHandler`, `CoreCollector`, `FileHelper` from `node-maintainer` to `node-admin`. (First 2 commits, no functional changes from original)
* Made some some simplification to those based on that this will not run nodes with yinst state
* Defined `CoredumpReporter` which can be implemented to handle authenticated and unauthenticated endpoints.

### CODE REVIEW

**Comment by baldersheim** [comment]
Copy, not move or shared ?

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/coredump/CoredumpReporter.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.yahoo.vespa.hosted.node.admin.maintenance.coredump;
+
+/**
+ * @author valerijf
```
Comment (hakonhall) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/coredump/CoredumpReporter.java]: freva

**Comment by hakonhall** [comment]
> Made some some simplification to those based on that this will not run nodes with yinst state

And we don't want them when host-admin rolls out on-prem?

**Comment by freva** [comment]
> And we don't want them when host-admin rolls out on-prem?

My understanding is that we will not use `yinst`, so listing yinst state is not needed?

**Comment by freva** [comment]
@hakonhall PTAL

**Review by hakonhall — APPROVED** [review]

**Pull request merged**